test/e2e-tests.sh

```diff
@@ -21,23 +21,25 @@ failed=0
 (( !failed )) && install_catalogsource || failed=1
 (( !failed )) && logger.success '🚀 Cluster prepared for testing.'
 
+# TEST_ALL used in nightly CI where all tests are run in one go
 if [[ $TEST_ALL == true ]]; then
```

Comment:
Curious why you still need the the TEST_ALL var in this script now that there's the `operator-tests.sh` script?

Reply:
Because the nightly run on OCP 4.3 and 4.4 runs all tests at once: upgrade, operator tests, serving, eventing. Without that we'd need to create one more script just for that purpose.

Reply:
With that being said, we could remove TEST_ALL and create one more script: `all-tests.sh`. That's what I was doing with my previous PR where @nak3 suggested using this TEST_ALL variable. Both solutions are possible.

Reply:
I was under the impression that e2e-tests.sh with TEST_ALL=false is basically what operator-tests.sh does.

Reply:
There are these levels (an Makefile targets):
* test-operator - only tests from serverless-operator repo
* test-e2e - only Serving, Eventing test suites (external), but without upgrades
* test-upgrade - only Serving upgrades
* test-all - all of the above together (this one is achieved by the TEST_ALL flag), but it could have a separate script too.

What's your suggestion?

Reply:
I think of any given repo as having two types of tests: unit (tests requiring only `go`) and e2e (tests requiring external infrastructure). Our operator has both, so it's a little confusing to have `test-operator` and `test-e2e` targets. The latter I might call `test-upstream` instead. And though I can see how it might be useful to treat `test-upgrade` as its own thing, I really see it as a subset of `test-upstream`.

So in terms of makefile targets, I was thinking:
- test-unit
- test-e2e
- test-upstream

All other targets could be defined in terms of those, e.g. `test-operator` would depend on the first two, and `test-all` would depend on all three. These would just be normal makefile deps between targets, so no need for any extra scripts.

At least, that's kinda how I was imagining it.

Reply:
So I was thinking you were using the TEST_ALL var as a way to distinguish between upstream e2e's and downstream (our) e2e's.

Reply:
Yeah. It makes sense. We just needs to distinguish upgrade tests and the other scenarios for CI where they need to run as separate Prow checks. I'm also not sure how the deps between targets will work in CI. The scripts will have to work when called separately and also when called one after another. I'll see how we can get closer to what you're suggesting and still satisfity ci-operator etc.

Reply:
so prow can run bash but not make?

Reply:
I've tried to address most of your concerns. Had to add some additional make targets for CI but overall I think this is pretty much what you wanted.